### PR TITLE
Space Manager fix

### PR DESCRIPTION
**Pull request by Crazylemon64**
The space manager now cleanly removes and adds space levels on demand

Also, space turfs will now re-gain their destination when over-written and re-created on the border.

:cl:Crazylemon
fix: Space border turfs will now regain their destination when over-written and replaced on the edge of the map
fix: The space manager will no longer runtime if a space ruin writes on the edge of the map
/:cl:

### CODE REVIEW

File: code/game/turfs/turf.dm [hunk 1]
```diff
@@ -186,11 +187,14 @@
...
 			lighting_clear_overlays()
-	
+
 	obscured = old_obscured
 
 	return W
 
+/turf/proc/PreChange()
+	return
+
```
Comment (Crazylemon64) [inline code/game/turfs/turf.dm]: I add this proc to answer the question of "What should a turf do before it is removed by `ChangeTurf`?"

Comment (Krausus) [inline code/game/turfs/turf.dm]: Shouldn't this be called `BeforeChange`, to go alongside `AfterChange`?

Also, there's probably some `ChangeTurf` overrides that need to become one or the other, such as the one on walls that deletes wallrot.

Comment (Crazylemon64) [inline code/game/turfs/turf.dm]: `Pre`, `Before`, they mean the same thing in this context, but `Before is probably a better antonym for `After` than `Pre` is.

And yeah, I'll move `ChangeTurf` overrides to this proc now, might as well

**Comment by TheDZD** [comment]
Conflicts.

**Pull request merged**